app/views/users/show.html.erb

```diff
@@ -8,4 +8,13 @@
       </h1>
     </section>
   </aside>
+  <div class="col-md-8">
+    <% if @user.microposts.any? %>
+      <h3><%= t ".micro"%>(<%= @user.microposts.count %>)</h3>
```

Comment:
so sánh size, count, length

Reply:
- cout: sẽ sd câu lệnh truy vấn sql để lấy ra số lượng bản ghi mà chúng ta truy vấn. cout không lưu trong object chúng ta gọi. mỗi lần gọi đến cout thì nó sẽ thực hiện câu lênh truy vấn vào database. dù trước đó chúng ta có gọi nó hay chưa.
- length: nó sẽ đếm số lượng mà không sử dụng thêm câu truy vấn nếu đối tượng đó đã được nạp, ngược lại, nó sẽ nạp toàn bộ object vào bộ nhớ và thực hiện đếm. nên khi gọi length từ lần thứ 2 trở đi nó sẽ không truy vấ vào databasse nữa mà sẽ trả về luôn kết quả.
- size: nó sẽ truy vấn vào database khi đối tượng chưa được nạp, còn đối tượng đã được nạp thì nó sẽ đếm luôn trong bộ nhớ chứ không truy vấn vào data base nữa